smack-core/src/main/java/org/jivesoftware/smack/altConnections/HttpLookupMethod.java

```diff
@@ -0,0 +1,174 @@
+            List<URI> endpoints = parseXrdLinkReferencesFor(xmlPullParser, relation);
+            return endpoints;
+        }
+    }
+
+    /**
+     * Get remote endpoints for the given LinkRelation from host.
+     *
+     * @param xmppServiceAddress address of host
+     * @param relation {@link LinkRelation} specifying type of connection
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> lookup(DomainBareJid xmppServiceAddress, LinkRelation relation) throws IOException, XmlPullParserException, URISyntaxException {
+        return lookup(xmppServiceAddress, relation.attribute);
+    }
+
+    /**
+     * Constructs a HTTP connection with the host specified by the DomainBareJid
+     * and retrieves XRD element in the form of an InputStream.
+     *
+     * @param xmppServiceAddress address of host
+     * @return InputStream containing XRD element
+     * @throws IOException exception due to input/output operations
+     */
+    public static InputStream getXrdStream(DomainBareJid xmppServiceAddress) throws IOException {
+        final String metadataUrl = "https://" + xmppServiceAddress.toString() + "/.well-known/host-meta";
+        final URL putUrl = new URL(metadataUrl);
+        final URLConnection urlConnection = putUrl.openConnection();
+        return  urlConnection.getInputStream();
+    }
+
+    /**
+     * Get remote endpoints for the provided LinkRelation from provided XmlPullParser.
+     *
+     * @param parser XmlPullParser that contains LinkRelations
+     * @param relation type of endpoints specified by the given LinkRelation
+     * @return list of endpoints
+     * @throws IOException exception due to input/output operations
+     * @throws XmlPullParserException exception encountered during XML parsing
+     * @throws URISyntaxException exception to indicate that a string could not be parsed as a URI reference
+     */
+    public static List<URI> parseXrdLinkReferencesFor(XmlPullParser parser, String relation) throws IOException, XmlPullParserException, URISyntaxException {
+        final String xrd_namespace = "http://docs.oasis-open.org/ns/xri/xrd-1.0";
+
+        List<URI> uriList = new ArrayList<>();
+        int initialDepth = parser.getDepth();
+
+        loop: while (true) {
+            XmlPullParser.TagEvent tag = parser.nextTag();
+            switch (tag) {
+            case START_ELEMENT:
+                String name = parser.getName();
+                String namespace = parser.getNamespace();
+                String rel = parser.getAttributeValue("rel");
+
+                if (!namespace.equals(xrd_namespace) || !name.equals("Link") || !rel.equals(relation)) {
+                    tag = parser.nextTag();
```

Comment:
That appears wrong now, as there is a call to `nextTag()` at the start of the loop.